### DISCUSSION TITLE
How do I handle POST, DELETE and PUT requests?

### DISCUSSION THREAD

**Discussion by coryfail** [TanStack/query#1767] [Q&A] [answered]
I've looked everywhere and I can't seem to find a good example of how to use React Query with POST, DELETE, and PUT requests. Is it because I just need to do a normal fetch without the use of React Query and then React Query detects a change and displays the fresh data?

**Comment by TkDodo** [comment, marked as answer]
those are `mutations`. Please see the [docs on mutations](https://react-query.tanstack.com/guides/mutations#_top) as well as the mutation examples: https://react-query.tanstack.com/examples/optimistic-updates

**Reply by Dal-Tron** [reply]
Hey, I'm just wondering if there is an example using a DELETE request I could look at?

**Reply by TkDodo** [reply]
mutations work exactly the same with post/put/patch/delete. There is no difference from RQ perspective because the lib is agnostic to how you do it.

**Comment by gatinhap** [comment]
> Hey, I'm just wondering if there is an example using a DELETE request I could look at?

@Dal-Tron, here is a simple example - https://github.com/gatinhap/plant-app-ver_0/blob/main/src/pages/removePlantPage/RemovePlant.tsx :)
